pkg/scrape/manager.go

```diff
@@ -113,16 +116,6 @@ func NewManager(
 	}
 	m.scrapeConfigs = c
 
-	// Cleanup and reload pool if config has changed.
-	for name, sp := range m.scrapePools {
```

Comment:
I have removed this block as I think it does nothing, let me know if I am wrong and I'll add it back.

It seems that at this point `m.scrapePools` is still an empty slice. And `m.ApplyConfig()` is called after `NewManager()` which itself calls `sp.reload()`